File: output/xCodeEval_remove_8000/C++/Go/640dc614f856ecc5cfc468b141ab77d4.go
package main

import (
	"fmt"
	"math"
	"time"
)

type pii struct {
	first  int64
	second int64
}

var TimeStart time.Time
var TimeEnd time.Time

var x int64

func func1(m, n, s int64) pii {
	if m == 0 {
		return pii{n, s}
	}
	k := int64(math.Cbrt(float64(m)))
	x := k * k * k
	y := (k - 1) * (k - 1) * (k - 1)
	return max(func1(m-x, n+1, s+x), func1(x-y-1, n+1, s+y))
}

func Input() {
	fmt.Scan(&x)
}

func Solve() {
	Token := func1(x, 0, 0)
	fmt.Println(Token.first, Token.second)
}

func main() {
	Input()
	TimerStart()
	Solve()
	TimerStop()
}

func TimerStart() {
	TimeStart = time.Now()
}

func TimerStop() {
	TimeEnd = time.Now()
	ElapsedTime := TimeEnd.Sub(TimeStart)
	fmt.Printf("\n\nTime elapsed: %f seconds.\n", ElapsedTime.Seconds())
}

func max(a, b pii) pii {
	if a.first > b.first {
		return a
	} else if a.first < b.first {
		return b
	} else {
		if a.second > b.second {
			return a
		} else {
			return b
		}
	}
}

File: output/xCodeEval_remove_8000/C++/C/640dc614f856ecc5cfc468b141ab77d4.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>
#include <time.h>

typedef long long i64;
typedef long double ld;

typedef struct {
    i64 first;
    i64 second;
} pii;

pii mp(i64 a, i64 b) {
    pii p;
    p.first = a;
    p.second = b;
    return p;
}

i64 max(i64 a, i64 b) {
    return (a > b) ? a : b;
}

i64 cbrt(i64 x) {
    return (i64)round(cbrtl((long double)x));
}

i64 x;

pii func(i64 m, i64 n, i64 s) {
    if (m == 0) return mp(n, s);
    i64 k = cbrt(m);
    i64 x = k * k * k, y = (k-1)*(k-1)*(k-1);
    return max(func(m-x, n+1, s+x), func(x-y-1, n+1, s+y));
}

void Input() {
    scanf("%lld", &x);
}

void Solve() {
    pii Token = func(x, 0, 0);
    printf("%lld %lld\n", Token.first, Token.second);
}

int main() {
    Input();
    Solve();
    return 0;
}
